uaclient/clouds/gcp.py

```diff
@@ -40,3 +48,9 @@ def is_viable(self) -> bool:
                 return True
 
         return False
+
+    def get_licenses_from_identity(self):
+        token = self.identity_doc["identityToken"]
+        identity = base64.urlsafe_b64decode(token.split(".")[1] + b"===")
+        identity = json.loads(identity)
+        return identity["google"]["compute_engine"]["license_id"]
```

Comment:
Maybe we should be a little more defensive when accessing those fields. I know that we surrounding this method in a try except block, but maybe we should only return an empty dict here instead.

Also, I think we should add code documentation and type hints for this method